### PR TITLE
Easymotion new features

### PR DESCRIPTION
**Pull request by MaxfieldWalker**
This PR includes some new features and improvements of easymotion.

### New Features

- n-character move `<leader><leader>/<character>...`
- two-chars move like `<leader><leader>2s<character><character>`

### Improvements

- Jump immediately if the only one candidate found to jump
- Add tests

This PR is a continuation of my previous PR #1932.

### CODE REVIEW

**Comment by Chillee** [comment]
Awesome!

**Comment by Chillee** [comment]
I'm going to update this branch because I fixed some issues with flaky tests on master.

**Comment by MaxfieldWalker** [comment]
Yes, I think it is ok to merge only this one or merge the previous one first, and then this one.

**Comment by MaxfieldWalker** [comment]
Unfortunately, ci-test seemed to fail. I will investigate the reason.

**Comment by Chillee** [comment]
I think you're good. The first 5 tests seem to be failing on master for w.e. reason as well.

**Comment by Chillee** [comment]
So to confirm, this is A-OK to merge?

**Comment by MaxfieldWalker** [comment]
Yes!

**Comment by Chillee** [comment]
Thank you for your great work! Could you just explain the thing brought up in the code review?

**Comment by MaxfieldWalker** [comment]
You mean I explain what is changed with this PR in the code review area on GitHub?

image

**Comment by Chillee** [comment]
@MaxfieldWalker Just specifically, what is `<leader><leader> 2s <char><char>`?

I don't see it on the easymotion github readme.

**Comment by MaxfieldWalker** [comment]
The difference between 's' and '2s' is characters used for search.
'2s' requires two characters to search.

Like the GIF below, <leader><leader>'s'<a> shows every line as candidate,
while <leader><leader>'2s'<a><b> shows lines as candidates without the first line because the first line contains only "a".

**Comment by Chillee** [comment]
@MaxfieldWalker Is this a standard vim-easymotion feature?

**Comment by MaxfieldWalker** [comment]
Yes, two-character search motion is explained [here](https://github.com/easymotion/vim-easymotion#2-character-search-motion).

**Comment by Chillee** [comment]
@MaxfieldWalker Yep our command system isn't expressive enough to express "unmapped" commands.

I was just wondering if there was a "semi-standard" mapping for those commands. Could you also explain them in the README?

File: README.md [hunk 1]
```diff
@@ -389,15 +389,18 @@ Once easymotion is active, you can initiate motions using the following commands
...
-`<leader> <leader> s <char>`|Search character
-`<leader> <leader> f <char>`|Find character forwards
-`<leader> <leader> F <char>`|Find character backwards
-`<leader> <leader> t <char>`|Til character forwards
-`<leader> <leader> T <char>`|Til character backwards
-`<leader> <leader> w`|Start of word forwards
-`<leader> <leader> b`|Start of word backwards
-`<leader> <leader> e`|End of word forwards
-`<leader> <leader> g e`|End of word backwards
+`<leader><leader> s <char>` or `<leader><leader> 2s <char><char>`|Search character
```
[the remark this answers is not in the document]
Comment (MaxfieldWalker) [reply README.md]: At present, count is not supported. '2' means just requiring two characters for search.
Original easymotion plugin seems not to support `<count>` for this kind of motions.

**Comment by MaxfieldWalker** [comment]
@Chillee I see.

**Comment by Chillee** [comment]
Thanks for the great work!

**Comment by Chillee** [comment]
ah damn I wasn't trying to squash and merge...

**Pull request merged**